Cristian Garcia Sprint Security Theory

2940cristian opened the pull request.

2940cristian committed:
Added regex portions

In theory section, completed the 3 regex exercises

2940cristian committed:
Security ReadMe part 1 done

Gave a few examples of each for threat model

2940cristian committed:
Fake email done

Used the examples from security readme and set up a scenario

2940cristian committed:
Practical Security done

Made two paragraphs about the actions I took to increase my security

2940cristian committed:
Changed first regex

Updated regex as Alex said it has to match the entire sentence

2940cristian committed:
Added lion state machine

2940cristian committed:
Added regex state machine

2940cristian committed:
VT100 regex complete

Added both regex solutions into theory readme

2940cristian committed:
Added VT100 state machine

cwindeknecht commented:
## Commit Guideline
### I have generalized these as opposed to giving you personalized commit critiques.  Surely you can discern for yourselves which of these recommendations you have or have not been following.  
#### If you are already following all of them, good on ya.  If you aren't, for your sakes to be prepared in the professional world / be able to pinpoint where in code you need to look to see what changes you made more easily, please begin to do so.
- Commit constantly!  Don't go overboard and do it with every line of code, but any time you finish a component, write an essential function, etc. just throw a commit up there. 
- Make sure to use a title and body.  If you are doing this from the command line, it will look like this: -m "Title" -m "Body".
- Think of what the title should be similar to if you were writing a book.  "part 1 done" doesn't tell you much unless you take the time to go find out what part 1 entailed.  Instead "Function for Dynamic File Reading Completed" is pretty obvious what you did.
- While body messages aren't always necessary, it is still usually a good idea to include them anyways.  From the above example, an ideal message would be "filename.js --- used readFile and regex to parse the program correctly."  Is that necessary?  Not really.  Later on, if you forgot where you were doing the dynamic file loading and you had that message, you wouldn't have to dig through the files to find where it was.  That doesn't really apply to our tiny programs that have a few files, but if you have way more it can really help. Think back to the front-end project for instance.  
- In general, you should only commit working code.  If you like to self-document what you did wrong to better your understanding, then a broken commit here and there isn't the end of the world.  I would suggest though, that if you want to do this, title it something akin to "Unfinished Dynamic File Reading" or "Broken Dynamic File Reading".  

## Code Review
### Good Work
* Well written security portion.  The regex / vt100 state machine was well done for the most part.  The regex technically fit the requirements, though I have some suggestions (see below).
* Super awesome commits!
### Work On  
* The lion state machine was a little jumbled, and some of the lines between states seemed disjointed (idle -> walking had lines labeled hungry / full).
* The only qualm with the regex state machine is the accept point isn't labeled as such (double circle signifies accept point).
* For the vt100 state machine, aside from the same error with the accept point from above, since it is representing a terminal, a line should be drawn from the accept point back to the start.  Otherwise it wouldn't properly represent a terminal that constantly accepts input.
* The antelope regex works, but I would suggest being more discretionary on what it takes.  For instance, 'antefdsfdssflopes rocfffdsks out' is accepted.  Similarly to how you just typed in out, you could have done something like 
```
antelopes? rocks? out
```
* The goat/moat/boat regex works, but 
1. The carrot at the beginning is extraneous.
2. You could have saved a couple of keystrokes with
```
[gm]oat
```

Overall, good Sprint Challenge, MVP achieved.

2940cristian committed:
Updated threatModel

Made changes to better reflect the situation and how it could be avoided in the future

cwindeknecht commented:
## Commit Guideline
### I have generalized these as opposed to giving you personalized commit critiques.  Surely you can discern for yourselves which of these recommendations you have or have not been following.  
#### If you are already following all of them, good on ya.  If you aren't, for your sakes to be prepared in the professional world / be able to pinpoint where in code you need to look to see what changes you made more easily, please begin to do so.
- Commit constantly!  Don't go overboard and do it with every line of code, but any time you finish a component, write an essential function, etc. just throw a commit up there. 
- Make sure to use a title and body.  If you are doing this from the command line, it will look like this: -m "Title" -m "Body".
- Think of what the title should be similar to if you were writing a book.  "part 1 done" doesn't tell you much unless you take the time to go find out what part 1 entailed.  Instead "Function for Dynamic File Reading Completed" is pretty obvious what you did.
- While body messages aren't always necessary, it is still usually a good idea to include them anyways.  From the above example, an ideal message would be "filename.js --- used readFile and regex to parse the program correctly."  Is that necessary?  Not really.  Later on, if you forgot where you were doing the dynamic file loading and you had that message, you wouldn't have to dig through the files to find where it was.  That doesn't really apply to our tiny programs that have a few files, but if you have way more it can really help. Think back to the front-end project for instance.  
- In general, you should only commit working code.  If you like to self-document what you did wrong to better your understanding, then a broken commit here and there isn't the end of the world.  I would suggest though, that if you want to do this, title it something akin to "Unfinished Dynamic File Reading" or "Broken Dynamic File Reading".  

## Code Review
### Good Work
* The regex / vt100 state machine was well done for the most part.  The regex technically fit the requirements, though I have some suggestions (see below).
* Super awesome commits!
### Work On  
* Could have been a bit more thorough on the security portion but what was there is good.
* The lion state machine was a little jumbled, and some of the lines between states seemed disjointed (idle -> walking had lines labeled hungry / full).
* The only qualm with the regex state machine is the accept point isn't labeled as such (double circle signifies accept point).
* For the vt100 state machine, aside from the same error with the accept point from above, since it is representing a terminal, a line should be drawn from the accept point back to the start.  Otherwise it wouldn't properly represent a terminal that constantly accepts input.
* The antelope regex works, but I would suggest being more discretionary on what it takes.  For instance, 'antefdsfdssflopes rocfffdsks out' is accepted.  Similarly to how you just typed in out, you could have done something like 
```
antelopes? rocks? out
```
* The goat/moat/boat regex works, but: 
1. The carrot at the beginning is extraneous.
2. You could have saved a couple of keystrokes with
```
[gm]oat
```
Good Sprint Challenge, MVP achieved.
